blocks/api/registration.js

```diff
@@ -73,6 +80,24 @@ export function registerBlockType( name, settings ) {
 		);
 		return;
 	}
+	if ( ! ( 'category' in settings ) ) {
+		console.error(
+			'The block "' + name + '" must have a category.'
+		);
+		return;
+	}
+	if ( 'category' in settings && typeof settings.category !== 'string' ) {
+		console.error(
+			'The category slug for block "' + name + '" must be a string.'
+		);
+		return;
+	}
+	if ( 'category' in settings && findIndex( categories, ( result ) => result.slug === settings.category ) === -1 ) {
```

Comment:
Also, the string check might be redundant with this here, since we know a non-string category would never match one of the known categories. But the more specific error message could be helpful (?)